packages/flutter_tools/bin/macos_assemble.sh

```diff
@@ -17,85 +17,124 @@ EchoError() {
   echo "$@" 1>&2
 }
 
-# Set the working directory to the project root
-project_path="${SOURCE_ROOT}/.."
-RunCommand pushd "${project_path}" > /dev/null
-
-# Set the verbose flag.
-verbose_flag=""
-if [[ -n "$VERBOSE_SCRIPT_LOGGING" ]]; then
-    verbose_flag="--verbose"
-fi
+BuildApp() {
```

Comment:
This logic is unchanged other than putting it in a function (so whitespace indentation) and adding `local`s.